elfp.c

```diff
@@ -144,8 +209,12 @@ test_one(char *f, Elf *elf, int flags)
 	goto out_print;
 
     /* arguably should print if P_OTHER, but, nah. */
-    if (ehdr.e_type != ET_DYN)
+    if (ehdr.e_type != ET_DYN) {
+	if ((flags & P_BUILDID) && b)
+	    goto out_print;
+
 	return;
```

Comment:
How do you manage to get here? What are you looking at that has a buildid but an unknown ELF type?